.github/workflows/tangle.yml

```diff
@@ -0,0 +1,46 @@
+name: LMT Tangle
+
+on:
+  push:
+    branches: 
+      - master
+    paths:
+      - 'Implementation.md'
+      - 'WhitespacePreservation.md'
+      - 'SubdirectoryFiles.md'
+      - 'LineNumbers.md'
+      - 'IndentedBlocks.md'
+
+jobs:
+  tangle:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Go setup
+        uses: actions/setup-go@v2
+        with: 
+          go-version: '^1.16'
+
+      - name: Build bootstrapping file
+        run: |
+          go build main.go
+          mv main lmt
+
+      - name: Tangle main using the bootstrapping file
+        run: ./lmt Implementation.md WhitespacePreservation.md SubdirectoryFiles.md LineNumbers.md IndentedBlocks.md
+
+      - name: Commit files
```

Comment:
I'm not sure I like the idea of GitHub actions commiting/pushing to the branch automatically and polluting the commit history.

What if, instead, this did something like:

```
mv main.go main.go.orig
./lmt Implementation.md WhitespacePreservation.md SubdirectoryFiles.md LineNumbers.md IndentedBlocks.md
cmp main.go main.go.orig
```

to verify that they're the same, and if not failed the build? (That also avoids the needs for the github secret token to be configured.)

Reply:
That might work, so long as lmt is rebuilt before every commit.